src/hex/grid.rs

```diff
@@ -123,15 +124,57 @@ impl<TileType> HexGrid<TileType> {
             }
         }
     }
+
+    /// Apply the shape onto the grid's collection.
+    ///
+    /// This will take every element inside a shape that is wrapped in Some()
+    /// and apply it into the grid's collection.
+    pub fn apply_shape(&mut self, shape: &HexShape<T>) -> &Self {
+        shape.get_hexes().indexed_iter().for_each(|ele| {
+            if let Some(value) = ele.1.as_ref() {
+                self.collection
+                    .insert(axial!(ele.0 .0 as i32, ele.0 .1 as i32), value.clone());
+            }
+        });
+        self
+    }
+
+    /// Extract data from the grid into the shape.
+    ///
+    /// Clones data from the grid's collection into the shape's internal working collection.
+    pub fn extract_shape(&self, shape: &mut HexShape<T>) -> &Self {
+        shape.get_hexes_mut().indexed_iter_mut().for_each(|ele| {
+            if ele.1.is_some() {
+                *ele.1 = self
+                    .collection
+                    .get(&axial!(ele.0 .0 as i32, ele.0 .1 as i32))
+                    .cloned();
+            } else {
+                *ele.1 = None;
+            }
+        });
+        self
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
-
     use crate::axial;
     use assert_float_eq::*;
 
+    #[test]
+    pub fn test() {
+        let mut grid = HexGrid::default();
+        grid.apply_shape(&HexShape::make_rhombus(8, 0, true, || 1));
+        println!("{:?}", grid.collection);
```

Comment:
i assume the final version is intended to test against the serialized version of the collection rather than printing it?

Reply:
Yes, that was just me making sure, via print statements, things were working as I expected them to as I developed the function.

Reply:
closing as this code is no longer here